### DISCUSSION TITLE
Customizing initializing and loading

### DISCUSSION THREAD

**Discussion by ovaisa** [iTwin/itwinjs-core#2421] [Q&A] [answered]
1) We have removed the sign-in/sign-out header bar on top of the Viewer and using our own login screens.  By removing the header bar, it seems there is now a gap between viewport bottom and status bar (please see attached image).  I tried to change height attribute in .viewer-container (in App.scss) but that doesn't seem to change anything.  Can you please guide

![StatusBar](https://user-images.githubusercontent.com/83007711/136029696-986d2125-7157-44b6-86cf-921ec158281e.JPG)

2) We would like to customize the loading spinner with our own branding.  I see that there are API functions to manage this (from frontend-showcase/UI-loading-icon example).  Would appreciate if I can get some sample code that shows how to apply this in the Viewer control.  I would like to use LoadingPrompt component with status displayed (last item in the UI-loading-icon example)

https://www.itwinjs.org/reference/ui-core/loading/

![Loading](https://user-images.githubusercontent.com/83007711/136030265-69e2decc-8c7f-42de-9cf5-b6e372ddbf17.JPG)

3) We would also like to customize the following message that appears when the iTwin.js icon (bottom left of the viewer) is clicked.  Any guidance on how to do this would be appreciated.

![PoweredBy](https://user-images.githubusercontent.com/83007711/136030480-03d7e2b7-a78c-4e74-a21a-7adceac86c07.JPG)

**Comment by pmconne** [comment]
The purpose of this dialog is to provide attribution for the contents of the viewer. For example, we are required to disclose the sources of map imagery and reality data obtained from certain third party sources like Bing and Cesium.
We also require applications to provide attribution to iModel.js.
You can add your own [application logo card](https://github.com/imodeljs/imodeljs/blob/098498f9ed2a52177b48ebe043ce25915cd5941a/core/frontend/src/IModelApp.ts#L617) to this dialog but you may not remove other information from it.

**Reply by kabentley** [reply]
also see [this article](https://www.itwinjs.org/learning/frontend/logocards/)

**Reply by ovaisa** [reply]
I was able to resolve (1) using scss from older version (v2.19.1) - I am using v2.19.6. for my app.  

App.scss (2.19.6)

.viewer-container {
  height: calc(100vh - 50px);
} 

I changed to the following from App.scss in v2.16.1 

.itwin-viewer-container {
  height: 100vh !important;
}

Originally 2.19.6 didn't contain the word 'itwin' but seems to accept and works correctly when I use .itwin-viewer-container

I changed v2.19.6 .viewer-container class to height: 100vh; but that didn't seem to affect the viewer height (tried other heights but with no affect).

Would appreciate confirmation if it is OK to use .itwin-viewer-container in 2.19.6 .  Thanks!

**Reply by aruniverse** [reply]
> I was able to resolve (1) using scss from older version (v2.19.1) - I am using v2.19.6. for my app.

> Originally 2.19.6 didn't contain the word 'itwin' but seems to accept and works correctly when I use .itwin-viewer-container

It sounds like you're referring to the two different templates, the deprecated `@bentley/itwin-viewer-react` which had the `itwin` prefix for the css class name, and the new one `@itwin/web-viewer-react` which doesn't have the `itwin` prefix for the class name. The templates have changed quite a bit between the two. Not sure which one you're using, or if you're using pieces from both, but your css change should be fine. 

Please don't confuse the version of iModel.js (you've mentioned 2.19.1, 2.19.6, 2.16.1) and the version of the iTwin Viewer. iTwin Viewer is built on top of iModel.js and is versioned separately. The version of iModel.js in your project needs to be in lockstep, as in all your core iModel.js dependencies need to be the same version. Just need to make sure they satisfy the requirements of the iTwin Viewer.  

> I changed v2.19.6 .viewer-container class to height: 100vh; but that didn't seem to affect the viewer height (tried other heights but with no affect).

You're probably missing the !important 
```scss
viewer-container { height: 100vh !important; }
```

**Reply by ovaisa** [reply]
Thank you Arun, adding '!important' to .viewer-container (as shown in your reply above) fixes the problem.

**Comment by aruniverse** [comment, marked as answer]
Re Q1:
It's hard for us to help you with scss changes directly if you don't show us the code and the changes you had made to replace the existing header.

Re Q2: 
Update your @itwin/viewer-react ver to the latest. A new prop has been added on the viewer, `loadingComponent`, for you to be able to replace the loader.
